src/transport/transport_dg.cc

```diff
@@ -205,16 +206,18 @@ void TransportDG<Model>::EqData::set_DG_parameters_boundary(Side side,
 template<class Model>
 TransportDG<Model>::TransportDG(Mesh & init_mesh, const Input::Record in_rec)
         : Model(init_mesh, in_rec),
+		  data_( make_shared<EqData>() ),
           input_rec(in_rec),
-          allocation_done(false)
+          allocation_done(false),
+		  multidim_assembly_new_{AssemblyDGNew<1, Model>(data_, *this), AssemblyDGNew<2, Model>(data_, *this), AssemblyDGNew<3, Model>(data_, *this)}
```

Comment:
Initialization provides correct inner AssemblyDGNew objects only in this constructor. In other methods these objects return unexpected results.